src/utils/components/CPUMemory/CPUMemory.tsx

```diff
@@ -19,17 +22,15 @@ const CPUMemory: FC<CPUMemoryProps> = ({ vm }) => {
   const { t } = useKubevirtTranslation();
   const itMatcher: V1InstancetypeMatcher = vm?.spec?.instancetype;
   const { instanceType, instanceTypeLoaded, instanceTypeLoadError } = useInstanceType(itMatcher);
+  const isVMRunning = vm?.status?.printableStatus === printableVMStatus.Running;
   const { vmi, vmiLoadError } = useVMI(getName(vm), getNamespace(vm));
 
-  if (vmiLoadError || (instanceTypeLoadError && !isEmpty(itMatcher)))
-    return <MutedTextSpan text={t('Not available')} />;
+  const cpu = vCPUCount(useCPUData(vm)) || instanceType?.spec?.cpu?.guest;
 
-  if (!vmi || !vm || !instanceTypeLoaded) return <Skeleton />;
+  if ((isVMRunning && vmiLoadError) || (!isEmpty(itMatcher) && instanceTypeLoadError))
+    return <MutedTextSpan text={t('Not available')} />;
 
-  const cpu =
-    vCPUCount(vmi?.spec?.domain?.cpu) ||
-    instanceType?.spec?.cpu?.guest ||
-    vCPUCount(vm?.spec?.template?.spec?.domain?.cpu);
+  if (!vm || !instanceTypeLoaded || (isVMRunning && !vmi)) return <Skeleton />;
 
   const memory = readableSizeUnit(
```

Comment:
u can add this to ur hook and also the instanceType call , just return the data

Reply:
We don't want this in "my" hook. Why? PTAl where it is used and why

Reply:
ur hook can be something to improve this whole modal as it will sperate between logic and display, if u move memory to there this component will only be in charge of rendering.